### DISCUSSION TITLE
Janusgraph could not deal with abnormal range queries for Predicate

### DISCUSSION THREAD

**Discussion by zhengyingying** [JanusGraph/janusgraph#2755] [Q&A] [answered]
For confirmed bugs, please report:
- Version: 0.5.3
- Storage Backend: inmemory
- Mixed Index Backend: none
- Link to discussed bug: 
- Expected Behavior: 
When we execute the query `g.E().has('ep2', inside(0.8480583995943037, 0.5952456882067648))`, we want search edges in which the value of property 'ep2' is within the range of the twp specified values. 
- Current Behavior:
Now, an exception `java.lang.illegalArgumentException` was throw. 
- Steps to Reproduce: 

### Stack Trace (if you have one)

```
java.util.concurrent.ExecutionException: org.apache.tinkerpop.gremlin.driver.exception.ResponseException: java.lang.IllegalArgumentException
```

**Comment by porunov** [comment]
Hi @zhengyingying , could it be related to #2273 ?
If so, are you able to test your case with the upcoming 0.6.0 release? You can download release artifacts here: https://github.com/JanusGraph/janusgraph/releases/tag/v0.6.0

**Comment by porunov** [comment, marked as answer]
@zhengyingying I converted the issue into a Discussion. I think this is most likely related to either https://github.com/JanusGraph/janusgraph/discussions/2754 (incorrect usage of Double values in Gremlin Console) or to #2273 .
Please, try using `g.E().has('ep2', inside(0.8480583995943037d, 0.5952456882067648d))` instead of `g.E().has('ep2', inside(0.8480583995943037, 0.5952456882067648))` and try to reproduce it on 0.6.0 JanusGraph release. In case it's reproducible, please, open a new issue with the reproduction scenario.

Also, your `inside` query logic isn't correct because you essentially saying `.and(P.gt(0.8480583995943037d), P.lt(0.5952456882067648d))` which will always result in false. Probably, you should swap your values.

**Reply by Mooner97** [reply]
@porunov Thanks. After testing, we ensure that this bug isn't reproducible in 0.6.0.
